lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -9036,11 +9032,23 @@ namespace Js
             length = pArr->length;
         }
 
+        if (length.IsSmallIndex())
```

Comment:
This method of factoring will invoke the large index version even for small indexes when the length itself is a large index. That might be a concern. This is, however, an edge case.